crypto/cms/cms_lib.c

```diff
@@ -39,15 +39,16 @@ CMS_ContentInfo *cms_Data_create(void)
 BIO *cms_content_bio(CMS_ContentInfo *cms)
 {
     ASN1_OCTET_STRING **pos = CMS_get0_content(cms);
-    if (!pos)
+
+    if (pos == NULL)
         return NULL;
     /* If content detached data goes nowhere: create NULL BIO */
-    if (!*pos)
+    if (*pos == NULL)
         return BIO_new(BIO_s_null());
     /*
      * If content not detached and created return memory BIO
      */
-    if (!*pos || ((*pos)->flags == ASN1_STRING_FLAG_CONT))
+    if (*pos == NULL || ((*pos)->flags == ASN1_STRING_FLAG_CONT))
```

Comment:
While you're at it, you could remove the extra parentheses

Reply:
no, because ```*``` is higher priority than ```->```
Would you prefer writing it as ```pos[0]->flags```?

Reply:
Don't remove parentheses as they aid readability

Reply:
I thought it was clear which ones would be extra.

``` C
    if (*pos == NULL || (*pos)->flags == ASN1_STRING_FLAG_CONT)
```

Reply:
The readability of the extra parenthesis is debatable.  In my view, they make it less readable...

Reply:
Tim wants them, so I guess I'll leave them to avoid his -1.
(FWIW, I agree they are superfluous and not helpful and redundant :)

Reply:
In this case, it becomes silly, with one sub-expression with extra parentheses and the other without.  The other consistent option would be

``` C
    if ((*pos == NULL) || ((*pos)->flags == ASN1_STRING_FLAG_CONT))
```

Reply:
I agree the line, as-is, looks silly.  I'll wait for you and Tim to resolve your difference over the diff.
:)

Reply:
The precedence of operators, which is among the first things a C apprentice should memorize, is neatly arranged in 'TABLE 2-1. Precedence and Associativity of Operators' of K&R (2nd edition). The table shows clearly that the `&&` and `||` have _much_ lower precedence than the boolean operators, it's almost just above the assignment operators. In fact, the precedence was chosen so low deliberately to avoid such silly parentheses. It's certainly not a corner case where you are never shure and parentheses help preventing mistakes. So +1 from me for removing them.

Reply:
Anyway, I guess @t-j-h's comment referred to the `(*pos)->flags` where the parentheses must not be omitted, and not to the ones around the boolean expression, correct?

Reply:
No. My comment was a general one. And I would add parentheses in both locations.
And it does aid readability - those who work in a cross-language environment or maintain larger code bases tend to add these items because they do aid readability - if you have to stop to think what the precedence rules are for a programming language in order to check some code then you are in essence directly supporting the concept that readability is reduced by leaving out parentheses. 

It is a fact of life that C programmers commonly make errors regarding the precedence rules of C in the real world ... and many many coding standards explicitly require use of (mostly redundant) parentheses to aid readability and maintainability.

And yes I do know views on this differ - but having had to fix many critic errors over the years because of people who got precedence rules incorrect, this is something I always put in place. Because they simply don't hurt to add - especially if you are consistent in their usage.

I don't -1 this sort of thing - but I do point out when someone asks to remove such things that removal of them isn't a requirement - as our coding standard is silent on the topic.

Reply:
Given the comment thread, I'll just leave it as-is unless directed otherwise.